pkg/resource/deploy/step.go

```diff
@@ -506,9 +508,18 @@ func (s *ReadStep) Apply(preview bool) (resource.Status, StepCompleteFunc, error
 			return resource.StatusOK, nil, err
 		}
 
-		result, err := prov.Read(urn, id, s.new.Inputs)
+		result, rst, err := prov.Read(urn, id, s.new.Inputs)
 		if err != nil {
-			return resource.StatusUnknown, nil, err
```

Comment:
Just so we are clear, the semantics of a partial failure of read are "The read found an object with the given ID and returned its state, but the object failed to reach a healthy state and `InitErrors` are the reasons why the object isn't healthy"?

Reply:
Yes. The same way this works in `Create` and `Update`.